### PR TITLE
Fixed noderunner issues on Windows by fixing a windows specific path...

### PR DESCRIPTION
**Pull request by Clintonio**
...quoting issue.

### CODE REVIEW

File: gradle-plugins/cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt [hunk 1]
```diff
@@ -64,16 +65,24 @@ private object WebJarType : JarType("corda-webserver.jar") {
...
 
     internal abstract fun processBuilder(): ProcessBuilder
     internal fun start() = processBuilder().directory(dir).start()
+
+    private val javaPath: String by lazy {
+        val path = File(File(System.getProperty("java.home"), "bin"), "java").path
+        // Replace below is to fix an issue with spaces in paths on Windows.
+        // Quoting the entire path does not work, only the space or directory within the path.
+        if(os == OS.WINDOWS) path.replace(" ", "\" \"") else path
+    }
```
Comment (andr3ej) [inline gradle-plugins/cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt]: Can you test whether this is needed in the --headless case?

**Comment by Clintonio** [comment]
Back in your court @andr3ej

**Comment by andr3ej** [comment]
Since the fix is not necessary in headless mode, I believe the real problem is in
```listOf("cmd", "/C", "start ${command.joinToString(" ")}")```
but this will do for now. Someone needs to work out how to quote things properly in the windows case, as is currently done in the unix case. The cmd.exe section of this webpage
https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
might be a good start.

**Comment by Clintonio** [comment]
Agree, though that looks pretty in depth though for this. I expect we'll find some issues with this fix down the line so this needs revisiting when I, or someone with a Windows machine/VM, has more time.

**Pull request merged**